vector/src/main/java/im/vector/app/features/matrixto/MatrixToBottomSheet.kt

```diff
@@ -44,10 +47,13 @@ class MatrixToBottomSheet :
 
     @Parcelize
     data class MatrixToArgs(
-            val matrixToLink: String
+            val matrixToLink: String,
+            val origin: OriginOfMatrixTo
     ) : Parcelable
 
     @Inject lateinit var avatarRenderer: AvatarRenderer
+    @Inject lateinit var session: Session
+    @Inject lateinit var appStateHandler: AppStateHandler
```

Comment:
Can be removed

Reply:
Here it's actually the only place where they are really used :)

Reply:
It's used because the analytics could be done in the navigator (like for the other cases), in the `mxToBottomSheetNavigateToRoom()` implementation. So I think it could be changed, and then these 2 lines could be removed.

What you need to do is to add the `trigger` parameter to `mxToBottomSheetNavigateToRoom()`, and pass it to the navigator.